src/lib/mongoBackend/connectionOperations.cpp

```diff
@@ -304,6 +319,7 @@ bool collectionInsert
   {
     connection->insert(col.c_str(), doc);
     releaseMongoConnection(connection);
+    TIME_STAT_MONGO_WRITE_WAIT_STOP();
```

Comment:
Maybe I'm missing something, but I understand that `TIME_STAT_MONGO_WRITE_WAIT_STOP()` should be also used after the `releaseMongoConnection(connection)` in the catch blocker where the program flow may end in the case of exception of conneciton insert.

I think the problem may happend in other functions in connectionOperations.cpp, as all them share the same try/catch structure.